mindsdb/integrations/handlers/merlion_handler/adapters.py

```diff
@@ -0,0 +1,162 @@
+    DEFAULT_MAX_BACKSTEP = 60
+    DEFAULT_MAXITER = 5
+
+    def __init__(self, **kwargs):
+        self.max_forecast_steps = kwargs.get(MerlionArguments.max_forecast_steps.value, self.DEFAULT_MAX_FORECAST_STEPS)
+        self.max_backstep = kwargs.get(MerlionArguments.max_backstep.value, self.DEFAULT_MAX_BACKSTEP)
+        self.maxiter = kwargs.get(MerlionArguments.maxiter.value, self.DEFAULT_MAXITER)
+        self.model = None
+
+    def to_bytes(self):
+        return self.model.to_bytes()
+
+    def from_bytes(self, bytes):
+        self.model = self.model.from_bytes(bytes)
+
+    def to_train_dataframe(self, df: pd.DataFrame, target: str) -> pd.DataFrame:
+        columns = list(df.columns.values)
+        columns.remove(target)
+        columns.insert(self.TARGET_SEQ_INDEX, target)
+        return df[columns]
+
+    def train(self, df: pd.DataFrame, target: str):
+        df = self.to_train_dataframe(df=df, target=target)
+        train_data = TimeSeries.from_pd(df)
+        self.model.train(train_data=train_data)
+
+    def predict(self, df: pd.DataFrame, target: str) -> pd.DataFrame:
+        forecast_step = self.max_forecast_steps
+        df = df[df.index <= self.model.last_train_time + self.model.timedelta * forecast_step]
+        if len(list(df.columns.values)) == 0:
+            df[target] = 0
+        predict_data = TimeSeries.from_pd(df)
+        predict_pred, predict_err = self.model.forecast(time_stamps=predict_data.time_stamps)
+        return self.__prepare_forecast_return(target=target, pred_ts=predict_pred, err_ts=predict_err)
+
+    def __prepare_forecast_return(self, target: str, pred_ts: TimeSeries, err_ts: TimeSeries) -> pd.DataFrame:
+        pred_df: pd.DataFrame = pred_ts.to_pd()
+        if err_ts is None:
+            err_df = None
+        else:
+            err_df = pred_ts.to_pd()
+
+        if err_df is None or target in list(err_df.columns.values):  # error and predict sometimes are same
+            std = pred_df[target].std()
+            pred_df[f"{target}__upper"] = pred_df[f"{target}"] + std * norm.ppf(0.975)
+            pred_df[f"{target}__lower"] = pred_df[f"{target}"] + std * norm.ppf(0.025)
+        else:
+            pred_df = pred_df.join(err_df, how="left")
+            pred_df[f"{target}__upper"] = pred_df[f"{target}"] + norm.ppf(0.975) * err_df[f"{target}_err"]
+            pred_df[f"{target}__lower"] = pred_df[f"{target}"] + norm.ppf(0.025) * err_df[f"{target}_err"]
+            pred_df.drop(columns=[f"{target}_err"], inplace=True)
+        return pred_df
+
+
+class DefaultForecasterAdapter(BaseMerlionForecastAdapter):
+    def __init__(self, **kwargs):
+        super(DefaultForecasterAdapter, self).__init__(**kwargs)
+        self.model = DefaultForecaster(DefaultForecasterConfig(max_forecast_steps=self.max_forecast_steps,
+                                                               target_seq_index=self.TARGET_SEQ_INDEX))
+
+
+class SarimaForecasterAdapter(BaseMerlionForecastAdapter):
```

Comment:
Maybe we can add doc string for each adapter and reference to the merlion docs for more info?